marklogic-spark-connector/src/test/java/com/marklogic/spark/writer/classifier/AddClassificationToJsonTest.java

```diff
@@ -71,6 +88,35 @@ void noClassificationAddedWhenNoSemaphoreServerSpecified() {
         assertFalse(doc.get("chunks").get(1).has("classification"));
     }
 
+    @Test
+    @EnabledIfEnvironmentVariable(named = "SEMAPHORE_API_KEY", matches = ".*")
```

Comment:
We want as few tests as possible to depend on Semaphore. Looking at the code, we can test everything except for actually invoking Semaphore. 

The following can be done in a follow-up PR:

1. Turn TextClassifier into an interface. 
2. Modify TextClassifierUdf to accept an instance of TextClassifier. 
3. Add tests that use a mock impl of TestClassifier, which returns a hardcoded response. 

That allows us to test the following key pieces:

1. How the response is added to the Spark row.
2. How the response is added to the 1 or more documents that are written. 

All future classifier tests should use that pattern so we can have as much coverage of the entire classifier feature without depending on an S4 connection.